clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -32,16 +33,39 @@ void processRecordDecl(RecordDecl *Declaration, ProgramInfo &Info,
       SourceManager &SM = Context->getSourceManager();
       FileID FID = FL.getFileID();
       const FileEntry *FE = SM.getFileEntryForID(FID);
+
+      //detect whether this RecordDecl is part of an inline struct
+      bool IsInLineStruct = false;
+      Decl *D = Declaration->getNextDeclInContext();
+      if (VarDecl *VD = dyn_cast_or_null<VarDecl>(D)) {
```

Comment:
Hm, now I am wondering if my interpretation of the semantics of `dyn_cast_or_null` is correct. Maybe it will just make `VD` NULL too and then execute the true branch? @jackastner do you know?

Reply:
Just checked the docs, you have it right :) 
> The dyn_cast_or_null<> operator works just like the dyn_cast<> operator, except that it allows for a null pointer as an argument (which it then propagates)